packages/db-objects-redis/src/lib/objects/objectsInRedisClient.js

```diff
@@ -2229,6 +2249,108 @@ class ObjectsInRedisClient {
+        }
+        if (Array.isArray(pattern)) {
+            let count = pattern.length;
+            pattern.forEach(pattern => {
+                const fileId = this.getFileId(id, pattern, false);
+                this.log.silly(`${this.namespace} redis psubscribe ${fileId}`);
+                subClient.psubscribe(fileId, err => {
+                    if (!err) {
+                        subClient.ioBrokerSubscriptions[fileId] = true;
+                    }
+                    if (!--count) {
+                        return tools.maybeCallbackWithError(callback, err);
+                    }
+                });
+            });
+        } else {
+            const fileId = this.getFileId(id, pattern, false);
+            this.log.silly(`${this.namespace} redis psubscribe ${fileId}`);
+            subClient.psubscribe(fileId, err => {
+                if (!err) {
+                    subClient.ioBrokerSubscriptions[fileId] = true;
+                }
+                return tools.maybeCallbackWithError(callback, err);
+            });
+        }
+    }
+
+    _unsubscribeFile(id, pattern, options, subClient, callback) {
+        if (!subClient) {
+            return tools.maybeCallbackWithRedisError(callback, ERRORS.ERROR_DB_CLOSED);
+        }
+        if (Array.isArray(pattern)) {
+            let count = pattern.length;
+            pattern.forEach(pattern => {
+                const fileId = this.getFileId(id, pattern, false);
+                this.log.silly(`${this.namespace} redis punsubscribe ${fileId}`);
+                subClient.punsubscribe(fileId, err => {
+                    if (!err && subClient.ioBrokerSubscriptions[fileId] !== undefined) {
+                        delete subClient.ioBrokerSubscriptions[fileId];
+                    }
+                    !--count && typeof callback === 'function' && callback(err);
+                });
+            });
+        } else {
+            this.log.silly(`${this.namespace} redis punsubscribe ${this.objNamespace}${pattern}`);
+            const fileId = this.getFileId(id, pattern, false);
+            subClient.punsubscribe(fileId, err => {
+                if (!err && subClient.ioBrokerSubscriptions[fileId] !== undefined) {
+                    delete subClient.ioBrokerSubscriptions[fileId];
+                }
+                typeof callback === 'function' && callback(err);
+            });
+        }
+    }
+
+    subscribeUserFile(id, pattern, options, callback) {
+        if (typeof options === 'function') {
+            callback = options;
+            options = null;
+        }
+        utils.checkObjectRights(this, null, null, options, 'list', (err, options) => {
+            if (err) {
+                return tools.maybeCallbackWithRedisError(callback, err);
+            } else {
+                return this._subscribeFile(id, pattern, options, this.sub, callback);
+            }
+        });
+    }
+
+    subscribeUserFileAsync(id, pattern, options) {
```

Comment:
Especially on DB level we can make the new stuff async only

Reply:
here i agree :-)

Reply:
Done